client/me/notification-settings/settings-form/stream.jsx

```diff
@@ -23,10 +22,7 @@ class NotificationSettingsFormStream extends PureComponent {
 		blogId: PropTypes.oneOfType( [ PropTypes.string, PropTypes.number ] ).isRequired,
 		stream: PropTypes.oneOfType( [ PropTypes.string, PropTypes.number ] ),
 		settingKeys: PropTypes.arrayOf( PropTypes.string ).isRequired,
-		settings: PropTypes.oneOfType( [
-			PropTypes.instanceOf( Immutable.List ),
-			PropTypes.instanceOf( Immutable.Map ),
-		] ).isRequired,
+		settings: PropTypes.object.isRequired,
```

Comment:
This prop type should be "array or object". When rendering settings for mobile devices, the `settings` prop is an array of object (one for each device). And it's just one object otherwise.

See also how the `getStreamSettings` method (few lines below) behaves differently when `props.devices` is present.

I found this bug when testing and seeing a prop type warning from React in console.